### PR TITLE
Create applicant account front end

### PR DESCRIPTION
**Pull request by SibingWu**
As per title
* Fixed issue #89 #104 
* Implement the front end for create applicant account feature
* Add the info page for user to create the applicant account
* Add non new user notification for applicant sign up/in

### CODE REVIEW

File: walk-in-interview/src/main/webapp/strings.en.js [hunk 1]
```diff
@@ -90,7 +90,10 @@ AppStrings['create-account'] = {
  * All the strings to be used in the account/create-account/applicant/index.html.
  */
 AppStrings['create-applicant-account'] = {
+  'name': 'Name',
+  'skills-title': 'Skills',
   'new-user-info': 'Account is successfully created!',
+  'non-new-user-info': 'This is a currently existing user. Log in successfully!',
```
Comment (yohanlaunay) [inline walk-in-interview/src/main/webapp/strings.en.js]: nit: "This is a currently existing user." <- it's a bit odd.  I don't think you've ever seen this in any website you logged into :) can you rephrase?

File: walk-in-interview/src/main/webapp/account/firebase-auth.js [hunk 2]
```diff
@@ -78,17 +80,13 @@ Auth.addPhoneSignInAndSignUpUI = (elementId, successPath, newUserInfo) => {
...
-          // Informs user
-          // TODO(issue/102): replace with proper notification
-          alert(newUserInfo);
+          onNewUser();
+        } else {
+          onExistingUser();
         }
 
-        return true;
+        return false;
```
Comment (yohanlaunay) [inline walk-in-interview/src/main/webapp/account/firebase-auth.js:91]: why the change? can you add  comment why "false" is important?

File: walk-in-interview/src/main/webapp/account/create-account/applicant/script.js [hunk 3]
```diff
@@ -51,13 +57,93 @@ function onLogOutFailure() {
...
 }
 
 const backButton = document.getElementById('back');
 backButton.addEventListener('click', (_) => {
-  window.location.href = LOGIN_HOMEPAGE_PATH;
+  window.location.href = CREATE_ACCOUNT_HOMEPAGE_PATH;
 });
+
+/**
+ * Creates a preliminary account object with phoneNumber as name for the user
```
Comment (yohanlaunay) [inline walk-in-interview/src/main/webapp/account/create-account/applicant/script.js:104]: please add `Returns blah blah` to the javadoc

File: walk-in-interview/src/main/webapp/account/create-account/applicant/script.js [hunk 4]
```diff
@@ -51,13 +57,93 @@ function onLogOutFailure() {
...
+      .then((response) => {
+        if (response.status == BAD_REQUEST_STATUS_CODE) {
+          setErrorMessage(/* errorMessageElementId= */'error-message',
+              /* msg= */ ACCOUNT_STRINGS['create-account-error-message'],
+              /* includesDefault= */false);
+          throw new Error(ACCOUNT_STRINGS['create-account-error-message']);
+        }
+
+        /** reset the error (there might have been an error msg from earlier) */
+        setErrorMessage(/* errorMessageElementId= */'error-message',
```
Comment (yohanlaunay) [inline walk-in-interview/src/main/webapp/account/create-account/applicant/script.js:134]: nit: i think you really need a `clearErrorMessage('error-message');` method. it looks like you are setting an error when you are actually clearing it.

File: walk-in-interview/src/main/webapp/account/create-account/applicant/account-info/script.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+  const accountDetails = getApplicantDetailsFromUserInput();
+
+  // Update the preliminary account object with more info
+  fetch(API['update-applicant-account'], {
+    method: 'POST',
+    headers: {'Content-Type': 'application/json'},
+    body: JSON.stringify(accountDetails),
+  })
+      .then((response) => {
+        if (response.status == BAD_REQUEST_STATUS_CODE) {
```
Comment (yohanlaunay) [inline walk-in-interview/src/main/webapp/account/create-account/applicant/account-info/script.js]: should you check for anything that's not 200 ?

**Review by yohanlaunay — CHANGES_REQUESTED** [review]

File: walk-in-interview/src/main/webapp/account/create-account/applicant/script.js (hunk 4 above)
Comment (SibingWu) [reply walk-in-interview/src/main/webapp/account/create-account/applicant/script.js:134]: will do in separate PR

**Review by yohanlaunay — APPROVED** [review]

**Review by htthach — APPROVED** [review]
